Pinta.Effects/Effects/VoronoiDiagramEffect.cs

```diff
@@ -0,0 +1,251 @@
+				typeof (DistanceMetric)),
+		};
+
+		static double Euclidean (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return difference.Magnitude ();
+		}
+
+		static double Manhattan (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return Math.Abs (difference.X) + Math.Abs (difference.Y);
+		}
+
+		static double Chebyshev (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return Math.Max (Math.Abs (difference.X), Math.Abs (difference.Y));
+		}
+	}
+
+	private static ImmutableHashSet<PointI> CreatePoints (RectangleI roi, int pointCount, RandomSeed pointLocationsSeed)
+	{
+		int effectivePointCount = Math.Min (pointCount, roi.Width * roi.Height);
+
+		Random randomPositioner = new (pointLocationsSeed.Value);
+		var result = ImmutableHashSet.CreateBuilder<PointI> (); // Ensures points' uniqueness
+
+		while (result.Count < effectivePointCount) {
+
+			PointI point = new (
+				X: randomPositioner.Next (roi.Left, roi.Right + 1),
+				Y: randomPositioner.Next (roi.Top, roi.Bottom + 1)
+			);
+
+			result.Add (point);
+		}
+
+		return result.ToImmutable ();
+	}
+
+	private static IEnumerable<ColorBgra> CreateColors (int colorCount, RandomSeed colorsSeed)
+	{
+		Random randomColorizer = new (colorsSeed.Value);
+		HashSet<ColorBgra> uniquenessTracker = new ();
+		while (uniquenessTracker.Count < colorCount) {
+			ColorBgra candidateColor = randomColorizer.RandomColorBgra ();
+			if (uniquenessTracker.Contains (candidateColor)) continue;
+			uniquenessTracker.Add (candidateColor);
+			yield return candidateColor;
+		}
+	}
+
+	public sealed class VoronoiDiagramData : EffectData
+	{
+		[Caption ("Distance Metric")]
+		public DistanceMetric DistanceMetric { get; set; } = DistanceMetric.Euclidean;
+
+		[Caption ("Number of Cells"), MinimumValue (1), MaximumValue (1024)]
+		public int NumberOfCells { get; set; } = 100;
+
+		// Translators: The user can choose whether or not to render the points used in the calculation of a Voronoi diagram
+		[Caption ("Show Points")]
+		public bool ShowPoints { get; set; } = false;
+
+		[Caption ("Color Sorting")]
+		public ColorSorting ColorSorting { get; set; } = ColorSorting.Random;
+
+		// Translators: In this context, "reverse" is a verb, and the user can choose whether or not they want to reverse the color sorting
+		[Caption ("Reverse Color Sorting")]
+		public bool ReverseColorSorting { get; set; } = false;
+
+		[Caption ("Random Colors")]
+		public RandomSeed RandomColors { get; set; } = new (0);
+
+		[Caption ("Random Point Locations")]
+		public RandomSeed RandomPointLocations { get; set; } = new (0);
+	}
+
+	public enum DistanceMetric
+	{
+		Euclidean,
+		Manhattan,
+		Chebyshev,
+	}
+
+	public enum ColorSorting
+	{
+		[Caption ("Random Color Sorting")] Random,
+
+		// Translators: Horizontal color sorting with B as the leading term
```

Comment:
For this group of comments, could you change phrases like `B` to something like `blue (B)`?
Translators might not be fully aware of the context of what the individual R/G/B characters mean here